SEED-platform/seed
# Add a cycle indicator to the analysis modal and filter analyses for current cycle

#### Any background context you want to provide?
Analyses and Cycles are linked through the AnalysisPropertyView table where there is a many-to-one relationship with Cycle. Currently, there is no indication which cycle an analysis is associated with and no filtering based on the relevant cycle.

#### What's this PR do?
- Adds a message to the "Create Analysis Modal" informing the user that the analysis will be associated with the current cycle.
- Adds cycle-specific analysis filtering to the Inventory Detail page's displayed analysis card.
- Adds cycle-specific analysis filtering to the Inventory Detail Analyses (beta).
- Adds a cycle selection dropdown to the Inventory Detail Analyses (beta) page.
- Adds a cycle indicator to the analysis card summary.
- Adds a cycle indicator to the analyses tables on the Analyses and Inventory Detail Analyses pages

#### How should this be manually tested?
This should be tested anywhere that an analysis may be created 
- Create an analysis from the Inventory List, ensure that cycle text is present and created cycle has a cycle indicator
- Create an analysis from the Inventory Detail, ensure that cycle text is present and created cycle has a cycle indicator. After creating the analysis it should appear in the top right of the Inventory Detail. Changing cycles will show only the most recent analysis for the selected cycle.
- Go to Analyses. Ensure there is a "Run Cycle" column in the analyses table. 
- Go to Property Detail Analyses (beta) page. Ensure there is a "Run Cycle" column in the analyses table. Click "New Analysis (beta)". Ensure there is informative text alerting the user to which cycle the analysis will be applied. 

#### What are the relevant tickets?
#3775 

#### Screenshots (if appropriate)

https://user-images.githubusercontent.com/58446472/217669445-576cbe08-8b95-4752-8af7-abbad7b71e13.mov


https://user-images.githubusercontent.com/58446472/217669458-77e9c9a2-a590-40c4-95cf-c7d7c794d0d2.mov

![analyses_detail_overview](https://user-images.githubusercontent.com/58446472/217669791-bb468d95-5f83-48b1-ab0d-820c352ce937.png)
![analyses_list_overview](https://user-images.githubusercontent.com/58446472/217669792-7b3b14e8-9d32-4c7f-bc7f-cf3a3498a3be.png)
![analysis_modal_overview](https://user-images.githubusercontent.com/58446472/217669794-eb7ae91e-449b-4578-9966-da2e72d67229.png)
